diff --git a/morello/src/codegen/mod.rs b/morello/src/codegen/mod.rs
index 7f3840c3..c17b08cd 100644
--- a/morello/src/codegen/mod.rs
+++ b/morello/src/codegen/mod.rs
@@ -31,7 +31,7 @@ pub use self::cpu::CpuCodeGenThreadStyle;
 const CLI_FLAGS: [&str; 4] = ["-std=gnu99", "-O3", "-rtlib=compiler-rt", "-o"];
 
 // TODO: Add -fopenmp if we're using an OpenMP pool.
-const X86_CLI_VEC_FLAGS: [&str; 2] = ["-mavx2", "-mfma"];
+const X86_CLI_VEC_FLAGS: [&str; 3] = ["-mavx2", "-mfma", "-fopenmp"];
 const ARM_CLI_VEC_FLAGS: [&str; 0] = [];
 const X86_MAC_HOST_CLI_VEC_FLAGS: [&str; 2] = ["-arch", "x86_64"];
 const ARM_MAC_HOST_CLI_VEC_FLAGS: [&str; 2] = ["-arch", "arm64"];
